tanner/utils/patterns.py

```diff
@@ -6,7 +6,9 @@
 LFI_FILEPATH = re.compile(r'((\.\.|/).*)')
 XSS_ATTACK = re.compile(r'.*<(.|\n)*?>')
 CMD_ATTACK = re.compile(
-    r'.*(alias|cat|cd|cp|echo|exec|find|for|grep|ifconfig|ls|man|mkdir|netstat|ping|ps|pwd|uname|wget|touch|while).*')
+    r'.*[^A-z:.]'
+    r'(alias|cat|cd|cp|echo|exec|find|for|grep|ifconfig|ls|man|mkdir|netstat|ping|ps|pwd|uname|wget|touch|while)'
+    r'[^A-z:.]?.*')
```

Comment:
Can you explain the use of `?` in this regexp. I think it shouldn't be present. Because we definitely don't want the presence of the set `[^A-z:.]` in the end of the matched keyword. `?` will make it a '0-1' case.
False-negative case for above exp:
`/news?color=https://www.evil.com/catalonia.php`
- Expected: No-Match
- Result: Match-found

Reply:
You are right :) My problem was in the end of the string, because command can be performed without  arguments. Now it should work